crates/sui-indexer/src/handlers/pruner.rs

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::time::Duration;
+
+use crate::{metrics::IndexerMetrics, store::IndexerStore, types::IndexerResult};
+
+pub struct Pruner<S> {
```

Comment:
i think we'd want to thread a cancellation token through here so we can also gracefully shut this service down